src/devices/CharacterLcd/GpioControllerAdapter.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Gpio;
+
+namespace Iot.Device.CharacterLcd
+{
+    public class GpioControllerAdapter : IGpioController
+    {
+        private GpioController _controller;
+
+        public GpioControllerAdapter(GpioController controller)
+        {
+            _controller = controller;
+        }
+
+        public void ClosePin(int pinNumber) => _controller.ClosePin(pinNumber);
+
+        public void Dispose() => _controller?.Dispose();
```

Comment:
We should probably also set _controller to null here so that 2 calls to dispose won't try to actually dispose twice.

Reply:
I can do that.

Reply:
Note though that there is no way to prevent double disposal without locking or doing an interlocked exchange.